commit 744d6f873393b6bbd12e1c1884738676a089fa65
Author: Simo Sorce <simo@redhat.com>
Date:   Sat Jul 20 13:20:43 2013 -0400

    Load import/export cred functions from GSS modules
    
    When the import/export credential feature was implement the related
    functions were added to struct gss_config, but the initialization
    function that dynamically loads modules was not changed to see if
    the plugin being loaded provided such functions.
    
    This will allow non-builtin mechanism and interposer mechanism to
    implement custom import/export credential extensions if they wish.
    
    ticket: 7682

diff --git a/src/lib/gssapi/mechglue/g_initialize.c b/src/lib/gssapi/mechglue/g_initialize.c
index 70cc4ee..48a825e 100644
--- a/src/lib/gssapi/mechglue/g_initialize.c
+++ b/src/lib/gssapi/mechglue/g_initialize.c
@@ -683,6 +683,8 @@ build_dynamicMech(void *dl, const gss_OID mech_type)
 	GSS_ADD_DYNAMIC_METHOD(dl, mech, gss_acquire_cred_from);
 	GSS_ADD_DYNAMIC_METHOD(dl, mech, gss_store_cred_into);
 	GSS_ADD_DYNAMIC_METHOD(dl, mech, gssspi_acquire_cred_with_password);
+	GSS_ADD_DYNAMIC_METHOD(dl, mech, gss_export_cred);
+	GSS_ADD_DYNAMIC_METHOD(dl, mech, gss_import_cred);
 	GSS_ADD_DYNAMIC_METHOD(dl, mech, gssspi_import_sec_context_by_mech);
 	GSS_ADD_DYNAMIC_METHOD(dl, mech, gssspi_import_name_by_mech);
 	GSS_ADD_DYNAMIC_METHOD(dl, mech, gssspi_import_cred_by_mech);
@@ -783,6 +785,8 @@ build_interMech(void *dl, const gss_OID mech_type)
 	RESOLVE_GSSI_SYMBOL(dl, mech, gss, _acquire_cred_from);
 	RESOLVE_GSSI_SYMBOL(dl, mech, gss, _store_cred_into);
 	RESOLVE_GSSI_SYMBOL(dl, mech, gssspi, _acquire_cred_with_password);
+	RESOLVE_GSSI_SYMBOL(dl, mech, gss, _export_cred);
+	RESOLVE_GSSI_SYMBOL(dl, mech, gss, _import_cred);
 	RESOLVE_GSSI_SYMBOL(dl, mech, gssspi, _import_sec_context_by_mech);
 	RESOLVE_GSSI_SYMBOL(dl, mech, gssspi, _import_name_by_mech);
 	RESOLVE_GSSI_SYMBOL(dl, mech, gssspi, _import_cred_by_mech);
